# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/src/prepare.rs
# Commit: e24ed89246a4
# Full Hash: e24ed89246a4a6c6b4a567ec412f0e3295b4570d
# Author: Glenn Watson <git@intuitionlibrary.com>
# Date: 2022-04-26 09:46:09
# Regressor Bug: 1766017
# File Overlap Count: 1
# Description:
#   Bug 1766017 - Do a better job of clipping off-screen surface allocation sizes r=gfx-reviewers,lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D144534
# ==============================================================================

diff -r cdc42b7822f1 -r e24ed89246a4 gfx/wr/webrender/src/prepare.rs
--- a/gfx/wr/webrender/src/prepare.rs	Mon Apr 25 22:28:35 2022 +0000
+++ b/gfx/wr/webrender/src/prepare.rs	Mon Apr 25 23:40:40 2022 +0000
@@ -761,7 +761,7 @@
                     let splitter = &mut frame_state.plane_splitters[plane_splitter_index.0];
                     let surface_index = pic.raster_config.as_ref().unwrap().surface_index;
                     let surface = &frame_state.surfaces[surface_index.0];
-                    let local_prim_rect = surface.local_rect.cast_unit();
+                    let local_prim_rect = surface.clipped_local_rect.cast_unit();
 
                     PicturePrimitive::add_split_plane(
                         splitter,